src/test/java/org/mule/service/http/impl/functional/server/HttpBrokenPipeLoggingConfigTestCase.java

```diff
@@ -0,0 +1,172 @@
+import org.junit.Before;
+import org.junit.Test;
+
+public class HttpBrokenPipeLoggingConfigTestCase extends AbstractHttpServerTestCase {
+
+  private CountDownLatch requestLatch;
+  private CountDownLatch responseLatch;
+  private ClassLoader requestHandlerClassLoader;
+
+  TestLogger testLogger = getTestLogger(BaseResponseCompletionHandler.class);
+
+  public HttpBrokenPipeLoggingConfigTestCase(String serviceToLoad) {
+    super(serviceToLoad);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    requestLatch = new CountDownLatch(1);
+    responseLatch = new CountDownLatch(1);
+
+    setUpServer();
+
+    // Changing the classloader in order to imitate the app's behaviour when handling the error
+    Thread currentThread = currentThread();
+    ClassLoader originalClassLoader = currentThread.getContextClassLoader();
+    requestHandlerClassLoader = CompositeClassLoader.from(currentThread().getContextClassLoader());
+    setContextClassLoader(currentThread, originalClassLoader, requestHandlerClassLoader);
+    try {
+      addRequestHandler();
+    } finally {
+      setContextClassLoader(currentThread, requestHandlerClassLoader, originalClassLoader);
+    }
+  }
+
+  @Override
+  protected String getServerName() {
+    return "broken-pipe-logging-test";
+  }
+
+  @Test
+  @Description("Verifies that HTTP broken pipe errors are logged in the app log for an app endpoint by checking the TCCL when the log was created")
+  public void brokenPipeErrorOnAppEndpointShouldBeLoggedOnTheAppLog() throws Exception {
+    try {
+      Request.Post(format("http://localhost:%s/%s", port.getValue(), "brokenPipe"))
+          .bodyString("That is not dead which can eternal lie, And with strange aeons even death may die.", DEFAULT_TEXT)
+          .socketTimeout(1)
+          .addHeader(CONNECTION, CLOSE)
+          .execute()
+          .returnResponse();
+      fail();
+    } catch (SocketTimeoutException ste) {
+      // Expected
+      requestLatch.countDown();
+    }
+
+    responseLatch.await();
+
+    new PollingProber(getTestTimeoutSecs(), 100).check(new JUnitLambdaProbe(() -> {
+      assertThat(testLogger.getAllLoggingEvents().size(), is(1));
+      assertThat(testLogger.getAllLoggingEvents().get(0).getThreadContextClassLoader(), is(requestHandlerClassLoader));
+      return true;
+    }));
+  }
+
+  @Test
+  @Description("Verifies that HTTP broken pipe errors are logged in the runtime log for unmapped paths by checking the TCCL when the log was created")
+  public void brokenPipeErrorOnUnmappedEndpointShouldBeLoggedOnTheRuntimeLog() throws Exception {
+    try {
+      Request.Post(format("http://localhost:%s/%s", port.getValue(), "unmapped"))
+          .bodyString("That is not dead which can eternal lie, And with strange aeons even death may die.", DEFAULT_TEXT)
+          .socketTimeout(1)
+          .execute()
+          .returnResponse();
+      fail();
+    } catch (SocketTimeoutException ste) {
+      // Expected
+    }
```

Comment:
Maybe we can extract this code fragment which is similar to the one in the previous test to a new private method that receives the target endpoint as a parameter.